horovod/common/util.py

```diff
@@ -251,3 +254,17 @@ def split_list(l, n):
     """
     d, r = divmod(len(l), n)
     return [l[i * d + min(i, r):(i + 1) * d + min(i + 1, r)] for i in range(n)]
+
+
+def check_installed_version(name, version, exception=None):
+    file_path = os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)),\
+        os.pardir, "metadata.json"))
+    with open(file_path) as f:
+        installed_version = json.load(f)[name]
```

Comment:
What if the user didn't build with this framework at all?

We should use `json.load(f).get(name)` to gracefully return `None`, otherwise the error message will be confusing.